src/gwt/src/org/rstudio/core/client/jsonrpc/RpcRequest.java

```diff
@@ -169,18 +171,47 @@ public void onResponseReceived(Request request,
                   String message = constants_.rpcErrorMessage(Integer.toString(status),
                           Desktop.isDesktop() ? constants_.rSessionMessage() : constants_.rStudioServerMessage(),
                           getMethod());
+
+                  int type = RpcError.TRANSMISSION_ERROR;
                   
                   // override error message for status code 0
                   if (status == 0)
                   {
-                     message = constants_.rpcOverrideErrorMessage((Desktop.isDesktop() ? constants_.rSessionMessage() : constants_.rStudioServerMessage()), getMethod());
+                     type = RpcError.TRANSMISSION_ERROR_NO_RESPONSE;
+                     if (Desktop.isDesktop())
+                     {
+                        message = constants_.rpcOverrideErrorMessage((Desktop.isDesktop() ? constants_.rSessionMessage() : constants_.rStudioServerMessage()), getMethod());
+                     } 
+                     else
+                     {
+                        // only show this error if not desktop and return early. Otherwise handle error as normal
+                        message = constants_.rpcOverrideErrorMessageServer((Desktop.isDesktop() ? constants_.rSessionMessage() : constants_.rStudioServerMessage()));
+                     
+                        if (!showingNoConnectError_) 
+                        {
+                           showingNoConnectError_ = true;
+                           RStudioGinjector.INSTANCE.getGlobalDisplay().showMessage(
+                              GlobalDisplay.MSG_ERROR,
+                              "RPC Error",
```

Comment:
Should add this to the `CoreClientConstants` string table so it can be localized.

Reply:
Will do.